src/coreComponents/mesh/mpiCommunications/MPI_iCommData.hpp

```diff
@@ -62,12 +62,12 @@ class MPI_iCommData
   /**
    * @return Reference to the field names registered with the communication data.
    */
-  std::map< string, string_array > const & getFieldNames() const { return m_fieldNames;}
+  std::vector< SyncFieldsID > const & getFieldsToBeSync() const { return m_fieldsToBeSync;}
 
   /**
    * @return Reference to the field names registered with the communication data.
    */
-  std::map< string, string_array > & fieldNames() { return m_fieldNames;}
+  std::vector< SyncFieldsID > & fieldsToBeSync() { return m_fieldsToBeSync;}
```

Comment:
I think you should define a real setter instead.
This solution is roughly equivalent to having a `public` variable.
But it also depends on the impact of the change...